### PR TITLE
Load E2E test secrets from env vars 

### PR DESCRIPTION
**Pull request by lukaszcl**
This is 1/2 PRs related to handing test secrets in Github CI. This PR adds most of the code to support loading secrets from env vars. But it still allow to have secrets in TOML. This will removed in a 2nd PR once QA Team migrates to env vars.

**User journey after migration**
1. Set secrets in a ~/.testsecrets file and put there test secrets as key=value
2. Install `ghsecrets` tool: `go install github.com/smartcontractkit/chainlink-testing-framework/tools/ghsecrets@latest` 
3. Run `ghsecrets set` to upload content of the ~/.testsecrets file to Github Secret Vault
4. Run tests on CI with `-f test_secrets_override_key={ghsecret_id}` to override default secrets in tests in a secure way 🔐 If `test_secrets_override_key` is not provided then default secrets set on CI will be used
5. You can still provide test config override with `-f test_config_override_base64=` flag  💥 

**Example ~/.testsecrets file**
```
CHAINLINK_IMAGE="http://url_to_qa_ecr_repo"
ARBITRUM_SEPOLIA_WALLET_KEY="testkey"
ARBITRUM_SEPOLIA_WALLET_KEY_1="key2"
ARBITRUM_SEPOLIA_RPC_HTTP_URL="http_url_2"
ARBITRUM_SEPOLIA_RPC_WS_URL_1="ws_url_1"
BSC_TESTNET_WALLET_KEY="key1"
BSC_TESTNET_RPC_HTTP_URL="http_1"
BSC_TESTNET_RPC_WS_URL="ws_1"
```

**Example of VRFV2Plus Smoke Test on CI**

- https://github.com/smartcontractkit/chainlink/actions/runs/9795862442/job/27049022325
- Has custom secrets like `BSC_TESTNET_WALLET_KEY` provided by Ilja and saved in `TEST_SECRETS_ILJA_PAVLOVS` secret in Github Secrets vault

**Example of Automation On Demand Tests on CI**
- https://github.com/smartcontractkit/chainlink/actions/runs/9796316824

**TODO merge related PRs**
- [x] https://github.com/smartcontractkit/chainlink-testing-framework/pull/1017
- [x] https://github.com/smartcontractkit/chainlink-github-actions/pull/183

### CODE REVIEW

File: .github/workflows/automation-ondemand-tests.yml [hunk 1]
```diff
@@ -134,7 +138,7 @@ jobs:
     env:
       CHAINLINK_COMMIT_SHA: ${{ github.sha }}
       CHAINLINK_ENV_USER: ${{ github.actor }}
-      TEST_LOG_LEVEL: info
+      TEST_LOG_LEVEL: debug
```
Comment (skudasov) [inline .github/workflows/automation-ondemand-tests.yml]: More logs in e2e tests are almost always better if QAs are fine with that

File: .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml [hunk 2]
```diff
@@ -46,10 +46,13 @@ jobs:
...
       - name: Run Tests
-        uses: smartcontractkit/chainlink-github-actions/chainlink-testing-framework/run-tests@af92c5fae8dcf1659201e907db82d221fc304b94 # v2.3.21
+        uses: smartcontractkit/chainlink-github-actions/chainlink-testing-framework/run-tests@cfda96316d2b0446815caba162b7bcbdf61765e7 # v2.3.21
         with:
-          test_command_to_run: cd ./integration-tests && go test -timeout 30m -count=1 -json -run ^TestVRFv2Plus$/^Link_Billing$ ./smoke/vrfv2plus_test.go 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false -hidepassinglogs
+          test_command_to_run: cd ./integration-tests && go test -timeout 30m -count=1 -json -run ^TestVRFv2Plus$/Native_Billing ./smoke/vrfv2plus_test.go 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false -hidepassinglogs
+          # test_command_to_run: cd ./integration-tests && go test -timeout 30m -count=1 -json -run ^TestVRFv2Plus$/^Link_Billing$ ./smoke/vrfv2plus_test.go 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false -hidepassinglogs
```
Comment (skudasov) [inline .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml]: Nitpick, please remove.

**Review by skudasov — DISMISSED** [review]

File: .github/workflows/automation-ondemand-tests.yml (hunk 1 above)
Comment (lukaszcl) [reply .github/workflows/automation-ondemand-tests.yml]: @skudasov I agree. But here I think I will go to `info`. I don't want to change Anirudh's log level

File: .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml (hunk 2 above)
Comment (lukaszcl) [reply .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml]: @skudasov yeah, I actually need to revert it. I was just using that test to test workflow with custom secrets (green build [here](https://github.com/smartcontractkit/chainlink/actions/runs/9795862442/job/27049022325) and more details in PR description)

File: .github/actions/setup-create-base64-config/action.yml [hunk 3]
```diff
@@ -85,12 +53,6 @@ runs:
         selected_networks=$(convert_to_toml_array "$SELECTED_NETWORKS")
         log_targets=$(convert_to_toml_array "$LOGSTREAM_LOG_TARGETS")
 
-        if [ -n "$PYROSCOPE_SERVER" ]; then
-          pyroscope_enabled=true
-        else
-          pyroscope_enabled=false
-        fi
```
Comment (Tofel) [inline .github/actions/setup-create-base64-config/action.yml:92]: why did you remove this logic? how will we decide now whether pyroscope should be enabled?

File: .github/workflows/on-demand-vrfv2-eth2-clients-test.yml [hunk 4]
```diff
@@ -46,10 +53,12 @@ jobs:
           echo "### Execution client used" >>$GITHUB_STEP_SUMMARY
           echo "\`${{ env.ETH2_EL_CLIENT }}\`" >>$GITHUB_STEP_SUMMARY          
       - name: Run Tests
-        uses: smartcontractkit/chainlink-github-actions/chainlink-testing-framework/run-tests@af92c5fae8dcf1659201e907db82d221fc304b94 # v2.3.21
+        uses: smartcontractkit/chainlink-github-actions/chainlink-testing-framework/run-tests@a0e61f5046fc157bc9a41f25f2e73fa726ec2474 # v2.3.21
         with:
           test_command_to_run: cd ./integration-tests && go test -timeout 30m -count=1 -json -run TestVRFv2Basic ./smoke/vrfv2_test.go 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false -hidepassinglogs
           test_download_vendor_packages_command: cd ./integration-tests && go mod download
+          test_config_override_base64: ${{ env.BASE64_CONFIG_OVERRIDE }}
+          test_secrets_override_base64: ${{ secrets[inputs.test_secrets_override_key] }}
```
Comment (Tofel) [inline .github/workflows/on-demand-vrfv2-eth2-clients-test.yml]: that's a nice one, I didn't know you access secrets like that 👍

File: .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml (hunk 2 above)
Comment (Tofel) [reply .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml]: @lukaszcl any reason for changing which test will be run?

File: integration-tests/testconfig/testconfig.go [hunk 5]
```diff
@@ -361,10 +367,10 @@ func GetConfig(configurationNames []string, product Product) (TestConfig, error)
 		logger.Debug().Msg("Base64 config override from environment variable not found")
 	}
 
-	// it needs some custom logic, so we do it separately
-	err := testConfig.readNetworkConfiguration()
+	logger.Info().Msg("Reading selected config values from environment variables")
+	err = testConfig.ReadConfigValuesFromEnvVars()
 	if err != nil {
-		return TestConfig{}, errors.Wrapf(err, "error reading network config")
+		return TestConfig{}, err
```
Comment (Tofel) [inline integration-tests/testconfig/testconfig.go]: why not wrap it?

File: .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml (hunk 2 above)
Comment (lukaszcl) [reply .github/workflows/on-demand-vrfv2plus-eth2-clients-test.yml]: @Tofel nope. This change will be reverted. Have a look at my comment in this thread ^

File: .github/workflows/on-demand-vrfv2-eth2-clients-test.yml (hunk 4 above)
Comment (lukaszcl) [reply .github/workflows/on-demand-vrfv2-eth2-clients-test.yml]: @Tofel yeah, this is quite elegant 🤩

File: integration-tests/testconfig/testconfig.go (hunk 5 above)
Comment (lukaszcl) [reply integration-tests/testconfig/testconfig.go]: @Tofel yeah, I will wrap it

**Review by gheorghestrimtu — APPROVED** [review]

**Review by skudasov — APPROVED** [review]
I'll approve it to unblock you @lukaszcl but I think we should add a chapter about how to extend secrets with a new variable. Sooner or later people will ask about it. Haven't found anything related here https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/config/README.md#test-secrets

**Review by jmank88 — APPROVED** [review]
